dbsc/workflow/rules/mrtpipelines.smk

```diff
@@ -1,571 +1,600 @@
             )
-        group: "subject_2"
-        containers:
-            config['singularity']['mrtpipelines'],
-        shell: 
-            'mrcalc -nthreads {params.threads} {input.subcortical_seg} 0 -neq {input.roi1} -sub {input.roi2} -sub {input.lZI} -sub {input.rZI} -sub {output.filter_mask} '
-            'tckedit -nthreads {params.therads} -exclude {output.filter_mask} -tck_weights_in {input.weights} -tck_weights_out {output.filtered_weights} {input.tck} {output.filtered_tck} '
-
-    rule combine_filtered:
-        input:
-            tck=expand(rules.filter_tck.output.filtered_tck, zip, **config['input_zip_lists']['T1w']),
-            weights=expand(rules.filter_tck.output.filtered_weights, zip, **config['input_zip_lists']['T1w']),
-        params:
-            threads=workflow.threads,
-        output:
-            combined_tck=bids(
-                root=join(config['out_dir'], 'mrtpipelines'),
-                datatype='tractography',
-                space='T1w',
-                desc='subcortical',
-                suffix='tractography.tck'
+        ),
+    threads: workflow.cores
+    group:
+        "subject_2"
+    container:
+        config["singularity"]["mrtrix"]
+    shell:
+        "mrcalc -nthreads {threads} {input.subcortical_seg} {wildcards.node1} -eq {output.roi_mask}"
+
+
+rule filter_tck:
+    # Node1 should be same as prev rule, need to iterate over node2
+    input:
+        roi1=bids(
+            root=mrtrix_dir,
+            datatype="anat",
+            desc="{node1}",
+            suffix="mask.mif",
+            **config["subj_wildcards"]
+        ),
+        roi2=bids(
+            root=mrtrix_dir,
+            datatype="anat",
+            desc="{node2}",
+            suffix="mask.mif",
+            **config["subj_wildcards"]
+        ),
+        # ZI rois (do these need to be separately defined in output?)
+        lZI=bids(
+            root=mrtrix_dir,
+            datatype="anat",
+            desc="21",
+            suffix="mask.mif",
+            **config["subj_wildcards"]
+        ),
+        rZI=bids(
+            root=mrtrix_dir,
+            datatype="anat",
+            desc="22",
+            suffix="mask.mif",
+            **config["subj_wildcards"]
+        ),
+        tck=rules.connectome2tck.output.edge_tck,
+        weights=rules.tck2connectome.outputs.node_weights,
+        subcortical_seg=rules.add_brainstem_new_seg.output.seg,
+    output:
+        filter_mask=temp(
+            bids(
+                root=mrtrix_dir,
+                datatype="anat",
+                desc="exclude",
+                suffix="mask.mif",
+                **config["subj_wildcards"]
             )
-            combined_weights=bids(
-                root=join(config['out_dir'], 'mrtpipelines'),
-                datatype='tractography',
-                space='T1w',
-                desc='subcortical',
-                suffix='tckweights.txt'
+        ),
+        filtered_tck=temp(
+            bids(
```

Comment:
Nitpick: The `desc` here isn't very BIDSy because it has a `-` and `_`. Maybe `desc=from{node1}to{node2}`?